### ISSUE TITLE
Can't set standalone presets in HA 2024.04.3

### ISSUE THREAD

**Issue by chriske** [jmcollin78/versatile_thermostat#444]
I'm trying to:
Set temperatures for not central managed thermostat.

![image](https://github.com/jmcollin78/versatile_thermostat/assets/3286943/df4351de-6202-4074-a9b4-05da6bd2e613)

As you can see, HA won't set it, because of a type issue. I tried with 23,0 and 23.0 but getting the same issue. Also tried to modify the values directly for the entities, but the issue is the same.

I installed your integration yesterday for 2024.04.2 and I was able to set presets for the central component. Today after the update I tried to create a new thermostat with it's own presets, but I run into this issue.

**Comment by jmcollin78** [comment]
Hello @chriske ,

I will try with 2024.04.2 to see if something change. It keep you in touch.
What is not normal in your screen copy, is that all temperature are empty. And because the error log says "NoneType" I guess you try to send an empty temperature on some Number entity.

Maybe you can try to put some value to all other temperature.

**Comment by chriske** [comment]
> Hello @chriske ,
> 
> I will try with 2024.04.2 to see if something change. It keep you in touch. What is not normal in your screen copy, is that all temperature are empty. And because the error log says "NoneType" I guess you try to send an empty temperature on some Number entity.
> 
> Maybe you can try to put some value to all other temperature.

Hi! Thanks for the response! I tried to populate every input but I'm getting the same issue for every one.

**Comment by jmcollin78** [comment]
Hello, I have tested with HA 2024.4.3 and all is working fine on my side. Something went wrong on your environment (all temperature are empty which is not possible). I suggest to check logs. There is surely some messages explaining what goes wrong.

 I will close the issue because it is not a VTherm issue but please post the logs messages here when you have.

**Comment by chriske** [comment]
Hi there!

Thanks for checking!

Here are the logs when I try to set any temperature:

2024-04-16 18:30:38.074 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [140282818477248] float() argument must be a string or a real number, not 'NoneType'
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 239, in handle_call_service
    response = await hass.services.async_call(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2543, in async_call
    response_data = await coro
                    ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/core.py", line 2580, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 971, in entity_service_call
    single_response = await _handle_entity_call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 1043, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/number/__init__.py", line 112, in async_set_value
    await entity.async_set_native_value(native_value)
  File "/config/custom_components/versatile_thermostat/number.py", line 462, in async_set_native_value
    old_value = float(self._attr_native_value)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: float() argument must be a string or a real number, not 'NoneType'

As far as I can see, the exception's origin is the number.py class

**Comment by chriske** [comment]
One more clue:

I tried to create a new VTherm, and now I choosed to not use the min and max temperatures from central config. Now my temperatures are populated and I can change them:

![image](https://github.com/jmcollin78/versatile_thermostat/assets/3286943/3c31296b-2042-4526-be6b-89a5ea0aaff5)

**Comment by jmcollin78** [comment]
Ooooh this is a great news.
What do you mean by " I choosed to not use the min and max temperatures from central config" ?
Can you send me screen copy of how you do that (because it seems related to the issue) ?

> I changed my previously posted VTherm's settings to not copy the central temperatures 

Does that means that you uncheck this checkbox (in french sorry) ?

![Capture d’écran 2024-04-17 à 08 15 40](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/aca72648-7f1e-4758-b3b4-8fe7ddbabdc3)

**Comment by chriske** [comment]
> Ooooh this is a great news. What do you mean by " I choosed to not use the min and max temperatures from central config" ? Can you send me screen copy of how you do that (because it seems related to the issue) ?
> 
> > I changed my previously posted VTherm's settings to not copy the central temperatures
> 
> Does that means that you uncheck this checkbox (in french sorry) ?
> 
> ![Capture d’écran 2024-04-17 à 08 15 40](https://private-user-images.githubusercontent.com/1712374/323101044-aca72648-7f1e-4758-b3b4-8fe7ddbabdc3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMzMzU2MjQsIm5iZiI6MTcxMzMzNTMyNCwicGF0aCI6Ii8xNzEyMzc0LzMyMzEwMTA0NC1hY2E3MjY0OC03ZjFlLTQ3NTgtYjNiNC04ZmU3ZGRiYWJkYzMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQxNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MTdUMDYyODQ0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2ZmYjBhNmY4NDA3MTc2ZWMwOTg1NzBkM2E1ZWY2NjMyMTcyNmIzNzIwZTQ3MTk1NDQ3Mzc3NGNiNTE0ZjA1MiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.dhoN1wPBINpWqTiMDPanZWwrV16tl_GCg1ZMxi7QMDU)

Exactly :) thanks!

**Comment by jmcollin78** [comment]
And for the min/max what have you done ?

**Comment by chriske** [comment]
These are my settings for the central configuration:

![image](https://github.com/jmcollin78/versatile_thermostat/assets/3286943/f4f0a070-425b-4737-84a3-902aeeffe31a)

**Comment by jmcollin78** [comment]
This was not my question. You said, you have done something with the min and max temperature. For me they are defined here: 
![Capture d’écran 2024-04-17 à 09 39 35](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/ea677607-6edc-4bc7-9f1f-1acb29dd4c78)

But maybe there is a misunderstanding somewhere.

**Comment by chriske** [comment]
Yes, it seems we misunderstood each other somewhere. So, I will summarize everything I have done:

First, I created a central configuration where I set up the outdoor thermometer, the minimum and maximum temperatures, and the step. To the presets created this way, I set the appropriate temperatures. After that, I made 2 VTherms, of the climate type, and set each to use the central configuration. These are working nicely. Following that, I created a third VTherm, for which I set that I wanted to define my own presets, but I left the "use additional central configuration" switch turned on in the underlyings menu. If the new VTherm is created this way, then I cannot set the temperatures for the presets, that’s when I get the above exception. If I turn off "use additional central configuration" afterwards, then I can set the temperatures in the presets created for the VTherm.

This is how it works:
![image](https://github.com/jmcollin78/versatile_thermostat/assets/3286943/7489a927-1f89-462f-aa29-b04cc1b1b8c0)

**Comment by brunoleon** [comment]
Hello,

I'm facing the same issue, and I did two videos to try to explain the behavior.

TLDR: if you do select "Use additional central main configuration" when creating a device, you won't be able to use device specific presets afterwards.

https://github.com/jmcollin78/versatile_thermostat/assets/2236444/3e9acabf-0137-4659-a259-ce0022099c65

https://github.com/jmcollin78/versatile_thermostat/assets/2236444/672d8488-579d-4174-ae52-496d2cdaed7a

**Comment by brunoleon** [comment]
Hello,

I'm facing the same issue, and I did two videos to try to explain the behavior.

TLDR: if you do select "Use additional central main configuration" when creating a device, you won't be able to use device specific presets afterwards.

https://github.com/jmcollin78/versatile_thermostat/assets/2236444/b3e2c143-552a-46e8-80ee-ad3244689e54


https://github.com/jmcollin78/versatile_thermostat/assets/2236444/7dadf402-e40e-4fe9-a906-79dbf264a316

**Comment by jmcollin78** [comment]
Thanks @brunoleon for the reproduction case. I will able to reproduce and to fix now !

**Comment by jmcollin78** [comment]
https://github.com/jmcollin78/versatile_thermostat/releases/tag/6.2.3

**Comment by chriske** [comment]
Thank you for the fix and your support!
